rusoto/core/src/signature.rs

```diff
@@ -93,10 +95,22 @@ impl SignedRequest {
         &self.headers
     }
 
+    pub fn scheme(&self) -> String {
+        match self.scheme {
+            Some(ref p) => p.to_string(),
+            None => {
+                match self.region {
+                    Region::Custom(_) => "http".to_owned(),
```

Comment:
Let's try it and find out!  I wonder if I can specify a "custom" location that's actually AWS to prove it works over HTTPS.